### PR TITLE
fixed warnings comes with fx cop and stylecop

### LINKED ISSUE #2: Fix warnings introduced with FX Cop and StyleCop
image

### PR DESCRIPTION
**Pull request by ajaygargdev**
Enable code analysis and address warnings and errors.

fixes #2

### CODE REVIEW

File: sample/Kmd.Logic.CitizenDocuments.Client.Sample/Kmd.Logic.CitizenDocuments.Client.Sample.csproj [hunk 1]
```diff
@@ -17,6 +17,10 @@
     <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="3.1.0" />
     <PackageReference Include="Serilog" Version="2.9.0" />
     <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
+    <PackageReference Include="StyleCop.Analyzers" Version="1.1.118">
```
Comment (adamchester) [inline sample/Kmd.Logic.CitizenDocuments.Client.Sample/Kmd.Logic.CitizenDocuments.Client.Sample.csproj]: Won’t this be inherited by the root directory.build.props file?

File: src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs [hunk 2]
```diff
@@ -44,22 +46,29 @@ public CitizenDocumentsClient(HttpClient httpClient, LogicTokenProviderFactory t
         /// <summary>
         /// Uploads the single citizen document.
         /// </summary>
+        /// <param name="configurationId">configuration id.</param>
+        /// <param name="retentionPeriodInDays">retention period.</param>
+        /// <param name="cpr">cpr.</param>
+        /// <param name="documentType">document type.</param>
+        /// <param name="document">document.</param>
+        /// <param name="documentName">document name.</param>
```
Comment (adamchester) [inline src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs]: Why bother repeating the parameter names in the documentation? Wouldn't it be better to either:

1. Remove the documentation entirely
2. Make it actually meaningful and useful

File: src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs [hunk 3]
```diff
@@ -70,10 +79,10 @@ public async Task<CitizenDocumentUploadResponse> UploadAttachmentWithHttpMessage
                     return null;
 
                 case System.Net.HttpStatusCode.Unauthorized:
-                    throw new CitizenDocumentsException("Unauthorized ", response.Body as string);
+                    throw new CitizenDocumentsException(response.Body.ToString(), response.Body as string);
```
Comment (adamchester) [inline src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs]: Why change this here?

File: src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs [hunk 4]
```diff
@@ -70,10 +79,10 @@ public async Task<CitizenDocumentUploadResponse> UploadAttachmentWithHttpMessage
...
 
                 default:
-                    throw new CitizenDocumentsException("Invalid configuration provided to access Citizen Document service", response.Body as string);
+                    throw new CitizenDocumentsException(response.Body.ToString(), response.Body as string);
```
Comment (adamchester) [inline src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs]: Why change this here?

File: sample/Kmd.Logic.CitizenDocuments.Client.Sample/Program.cs [hunk 5]
```diff
@@ -75,6 +74,7 @@ private static async Task Run(AppConfiguration configuration)
                     Log.Error("There is error occured in upload");
                     return;
                 }
+
                 Log.Information($"Document uploaded successfully and details are :-  DocumentId : {uploadDocument.DocumentId} ; DocumentType : {uploadDocument.DocumentType} ; FileAccessPageUrl : {uploadDocument.FileAccessPageUrl} ");
```
Comment (adamchester) [inline sample/Kmd.Logic.CitizenDocuments.Client.Sample/Program.cs]: Avoid using string interpolation with Serilog.

File: src/Kmd.Logic.CitizenDocuments.Client/Kmd.Logic.CitizenDocuments.Client.csproj [hunk 6]
```diff
@@ -8,6 +8,10 @@
 
   <ItemGroup>
     <PackageReference Include="Kmd.Logic.Identity.Authorization" Version="1.0.0" />
+    <PackageReference Include="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
```
Comment (adamchester) [inline src/Kmd.Logic.CitizenDocuments.Client/Kmd.Logic.CitizenDocuments.Client.csproj]: Shouldn't this be inherited from the `Directory.Build.props` file?

**Review by adamchester — CHANGES_REQUESTED** [review]
Left a few comments and questions.

File: src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs (hunk 2 above)
Comment (Ashish0410verma) [reply src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs]: Made param name description meaningful.

File: src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs (hunk 3 above)
Comment (Ashish0410verma) [reply src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs]: Sending message now

File: src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs (hunk 4 above)
Comment (Ashish0410verma) [reply src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs]: Sending message now

File: sample/Kmd.Logic.CitizenDocuments.Client.Sample/Program.cs (hunk 5 above)
Comment (Ashish0410verma) [reply sample/Kmd.Logic.CitizenDocuments.Client.Sample/Program.cs]: Ok. Removed string interpolation at all the places where logging events with serilog.

File: src/Kmd.Logic.CitizenDocuments.Client/Kmd.Logic.CitizenDocuments.Client.csproj (hunk 6 above)
Comment (Ashish0410verma) [reply src/Kmd.Logic.CitizenDocuments.Client/Kmd.Logic.CitizenDocuments.Client.csproj]: Was not sure before, but now removed from citizen document sample and client project. 👍

File: sample/Kmd.Logic.CitizenDocuments.Client.Sample/Kmd.Logic.CitizenDocuments.Client.Sample.csproj (hunk 1 above)
Comment (Ashish0410verma) [reply sample/Kmd.Logic.CitizenDocuments.Client.Sample/Kmd.Logic.CitizenDocuments.Client.Sample.csproj]: Was not sure before, but now removed from citizen document sample and client project. 👍

**Comment by adamchester** [comment]
@ajaygargdev it looks like this PR was superseded. If that's true, can you close this please?

**Pull request merged**